ext/case_transform/src/lib.rs

```diff
@@ -2,80 +2,124 @@
-        deepTransformKeys(value, |key| transformMethod(key))
+            result.store(new_key, new_value);
+        });
+
+        result.to_any_object()
     }
+}
+
+impl Transform for Array {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject {
+        // Temp hack to consume &self for iterator
+        let result = unsafe { self.to_any_object().to::<Array>() };
 
-    fn transformSymbol(value: Symbol, transformMethod: &Fn(AnyObject) -> AnyObject) -> Symbol {
-        let transformed = transformMethod(value);
-        Symbol::new(transformed);
+        result.into_iter()
+            .map(|item| transform(item, transform_function))
+            .collect::<Array>()
+            .to_any_object()
     }
+}
 
-    fn transform(
-        value: AnyObject,
-        objectTransform: &Fn(AnyObject) -> AnyObject,
-        keyTransform: &Fn(String) -> String
-    ) -> AnyObject {
-        match value.unwrap().ty() {
-            ValueType::Array => transformArray(value, objectTransform).to_any_object(),
-            ValueType::Hash => transformHash(value, objectTransform).to_any_object(),
-            ValueType::Symbol => transformSymbol(value, objectTransform).to_any_object(),
-            ValueType::RString => keyTransform(value).to_any_object(),
-            ValueType::Object => value
-        }
+trait TryTransform: Object {
+    fn try_transform<T>(&self,
+                        transform_function: &Fn(String) -> String)
+                        -> Result<AnyObject, RuruError>
+        where T: VerifiedObject + Transform
+    {
+        self.try_convert_to::<T>().map(|object| object.transform(transform_function))
     }
+}
+
+impl TryTransform for AnyObject {}
+
+fn transform(object: AnyObject, key_transform: &Fn(String) -> String) -> AnyObject {
+    let result = object.try_transform::<RString>(key_transform)
+        .or_else(|_| object.try_transform::<Symbol>(key_transform))
+        .or_else(|_| object.try_transform::<Array>(key_transform))
+        .or_else(|_| object.try_transform::<Hash>(key_transform))
+        .or_else(|_| object.try_transform::<AnyObject>(key_transform));
+
+    result.unwrap()
+}
+
+fn to_pascal_case(key: String) -> String {
+    classcase::to_class_case(snakecase::to_snake_case(key))
+}
+
+fn to_camel_case(key: String) -> String {
+    camelcase::to_camel_case(snakecase::to_snake_case(key))
+}
+
+fn to_dashed_case(key: String) -> String {
+    kebabcase::to_kebab_case(snakecase::to_snake_case(key))
+}
+
+fn to_snake_case(key: String) -> String {
+    snakecase::to_snake_case(key)
+}
 
-    fn toPascalCase(key: String) -> String { to_class_case(to_snake_case(key.unwrap())) }
-    fn toCamelCase(key: String) -> String { to_camel_case(to_snake_case(key.unwrap())) }
-    fn toDashedCase(key: String) -> String { to_kebab_case(to_snake_case(key.unwrap())) }
-    fn toSnakeCase(key: String) -> String { to_snake_case(key.unwrap()) }
+class!(CaseTransform);
+
+methods! (
```

Comment:
All the functions which are not used directly in the `CaseTransform` class are moved outside `methods!`